File: Striver A2Z/Step 3 - Solve Problems on Arrays [Easy->Medium->Hard]/Lec 2 - Medium/two-sum.cpp

// https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        int n = nums.size();
        if(n == 0) return ans;

        unordered_map<int, int> mp;
        for(int i = 0; i < nums.size(); i++){
            if(mp.count(target-nums[i]) > 0){
                ans.push_back(i);
                ans.push_back(mp[target-nums[i]]);
                return ans;
            }
            mp[nums[i]] = i;
        }
        return ans;
    }
};

File: Neetcode-150-js/Arrays and Hashing/two-sum.js

// https://leetcode.com/problems/two-sum/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = {};
    for(let i = 0; i < nums.length; i++){
        if(map[target - nums[i]] >= 0){
            return [i, map[target-nums[i]]];
        }
        map[nums[i]] = i;
    }
    return [];
};